heron/stmgr/tests/cpp/server/stmgr_unittest.cpp

```diff
@@ -1521,7 +1521,8 @@ TEST(StMgr, test_tmaster_restart_on_same_address) {
   // Note: Here we sleep longer compared to the previous test as we need
   // to tmasterClient could take upto 1 second (specified in test_heron_internals.yaml)
   // to retry connecting to tmaster.
-  sleep(3);
+  while (regular_stmgr->GetPhysicalPlan()->stmgrs(1).data_port() == common.stmgr_baseport_ + 1)
```

Comment:
It's better to provide a reasonable timeout here for how long the test should wait. The CI timeouts can be an hour or more. We don't want this test to hang for that long and potentially clog what is a resource-limited CI queue.